### PR TITLE
Add reserve system

### PR DESCRIPTION
**Pull request by neokry**
Allow for first n tokens to be reserved for claiming. useful for usecases like dao migrations or claim systems for existing collections

### CODE REVIEW

File: src/token/Token.sol [hunk 1]
```diff
@@ -407,7 +424,7 @@ contract Token is IToken, VersionedContract, UUPS, Ownable, ReentrancyGuard, ERC
         settings.totalOwnership = 0;
         emit FounderAllocationsCleared(newFounders);
 
-        _addFounders(newFounders);
+        _addFounders(newFounders, reservedUntilTokenId);
```
Comment (iainnash) [inline src/token/Token.sol:427]: we just have to make sure to update our UI to handle token gaps

File: src/token/storage/TokenStorageV3.sol [hunk 2]
```diff
@@ -0,0 +1,10 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.16;
+
+/// @title TokenStorageV3
+/// @author Neokry
+/// @notice The Token storage contract
+contract TokenStorageV3 {
+    /// @notice Marks the first n tokens as reserved
+    uint256 reservedUntilTokenId;
```
Comment (iainnash) [inline src/token/storage/TokenStorageV3.sol:9]: should this be `public`?

Comment (neokry) [reply src/token/storage/TokenStorageV3.sol]: updated to `public`

**Review by jgeary — APPROVED** [review]
LGTM

**Comment by jgeary** [comment]
@neokry please run the storage generate script so the new variable `reservedUntilTokenId` is included and fix failing tests.

**Pull request closed without merging**